quesma/v2/core/quesma_builder.go

```diff
@@ -45,11 +50,19 @@ func (quesma *Quesma) Stop(ctx context.Context) {
 }
 
 func (quesma *Quesma) Build() (QuesmaBuilder, error) {
+
+	var componentsToInjectDependencies []interface{}
+	deferInjectDepenencies := func(a any) {
+		componentsToInjectDependencies = append(componentsToInjectDependencies, a)
+	}
+
 	endpoints := make(map[string]struct{})
 	handlers := make(map[string]HandlersPipe)
 
 	for _, pipeline := range quesma.pipelines {
```

Comment:
I would prefer separate pass for dependency injection. In the future, we might have functionality like `registerPass` and then it would be easy to just copy paste this code.